src/test/java/unit/GcloudBuildsSubmit.java

```diff
@@ -0,0 +1,93 @@
+import bio.terra.cli.businessobject.Server;
+import com.google.cloud.Identity;
+import com.google.cloud.storage.BucketInfo;
+import harness.TestCommand;
+import harness.TestUser;
+import harness.baseclasses.SingleWorkspaceUnit;
+import harness.utils.Auth;
+import harness.utils.ExternalGCSBuckets;
+import java.io.File;
+import java.io.IOException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra gcloud builds submit` commands. */
+@Tag("unit")
+public class GcloudBuildsSubmit extends SingleWorkspaceUnit {
+
+  // external bucket to use for creating GCS bucket references in the workspace
+  private BucketInfo externalSharedBucket;
+  private TestUser shareUser;
+
+  @BeforeAll
+  @Override
+  protected void setupOnce() throws Exception {
+    super.setupOnce();
+    externalSharedBucket = ExternalGCSBuckets.createBucketWithUniformAccess();
+
+    // grant the user's proxy group access to the bucket so that it will pass WSM's access check
+    // when adding it as a referenced resource
+    ExternalGCSBuckets.grantWriteAccess(
+        externalSharedBucket, Identity.group(Auth.getProxyGroupEmail()));
+
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getUserFacingId());
+    shareUser = TestUser.chooseTestUserWhoIsNot(workspaceCreator);
+    TestCommand.runCommandExpectSuccess(
+        "workspace", "add-user", "--email=" + shareUser.email, "--role=WRITER");
+    shareUser.login();
+    ExternalGCSBuckets.grantReadAccess(
+        externalSharedBucket, Identity.group(Auth.getProxyGroupEmail()));
+
+    // create a dockerfile as building source
+    new File("./Dockerfile").createNewFile();
+  }
+
+  @AfterAll
+  @Override
+  protected void cleanupOnce() throws Exception {
+    super.cleanupOnce();
+    ExternalGCSBuckets.deleteBucket(externalSharedBucket);
+    externalSharedBucket = null;
+
+    // delete the dockerfile
+    new File("./Dockerfile").delete();
+  }
+
+  @Test
+  @DisplayName("builds submit --gcs-bucket")
+  void build() throws IOException {
+    Server server = Context.getServer();
+    if (!server.getCloudBuildEnabled()) {
+      return;
+    }
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getUserFacingId());
+
+    // `terra resource create gcs-bucket --name=$name --format=json`
+    String name = "resourceName";
+    TestCommand.runCommandExpectSuccess("resource", "create", "gcs-bucket", "--name=" + name);
+
+    // `builds submit --gcs-bucket=bucketName`
+    TestCommand.runCommandExpectSuccess("gcloud", "builds", "submit", "--gcs-bucket=" + name);
+
+    // `terra resource delete --name=$name`
+    TestCommand.Result cmd =
+        TestCommand.runCommand("resource", "delete", "--name=" + name, "--quiet");
+
+    boolean cliTimedOut =
```

Comment:
Why have this? Command shouldn't time out, right? Delete.

Reply:
Hi Melissa. The bucket is currently used by the gcloud building, so resource delete is blocked by waiting for the bucket to be released. During testing, I experienced time out error here, so I expect a succeed or time out here. Note: the resource will finally be guaranteed to clean up since workspace is deleted in the clean up step.

I also tried to set a timeout=1 second for the gcloud building, but the resource delete still get blocked.

Reply:
Can you point me to the build scan for the failure that you get, when this line is deleted?